web/src/pages/people/index.tsx

```diff
@@ -44,6 +54,40 @@ export default function PeoplePage() {
   const isAdmin = userData?.roles.includes("admin");
   const isIAO = userData?.roles.includes("information-asset-owner");
   const isTreOpsStaff = userData?.roles.includes("tre-ops-staff");
+  const canSearch = isTreOpsStaff || isAdmin;
+
+  const handleUserSearch = async (query: string) => {
+    if (query.length < 3) {
+      setSearchErrorMessage("minimum 3 characters required to perform search");
+      return;
+    }
+
+    if (query === "") {
+      setUsers(originalUsers);
+      return;
+    }
+
+    const regex = `^\w[\w.\s0-9@]+\w$`;
+    const isValid = new RegExp(regex).test(query);
+    if (!isValid) {
+      setSearchErrorMessage("invalid query");
+      return;
+    }
```

Comment:
I think the regex makes the length and empty string checks redundant right?

Reply:
ah yes, but now I'm getting page reloads when submitting, which isn't ideal 🤔

Reply:
It's because of the [prevent default](https://github.com/ucl-arc-tre/portal/pull/251#discussion_r2318508114)

Reply:
grazie
https://github.com/ucl-arc-tre/portal/pull/251/commits/231331464a434482fc8556637c94c2d3d32da4e1

Reply:
It looks like I am always hitting the invalid query error no matter what i type in now